stac_pydantic/api/search.py

```diff
@@ -102,32 +98,43 @@ def validate_bbox(cls, v: BBox) -> BBox:
 
     @field_validator("datetime")
     @classmethod
-    def validate_datetime(cls, v: str) -> str:
-        if "/" in v:
-            values = v.split("/")
-        else:
-            # Single date is interpreted as end date
-            values = ["..", v]
-
-        dates: List[dt] = []
-        for value in values:
-            if value == ".." or value == "":
-                continue
-
-            dates.append(parse_rfc3339(value))
+    def validate_datetime(cls, value: str) -> str:
```

Comment:
Would it be helpful to be more permissive here and accept 
`value: str | UtcDatetime | tuple[UtcDatetime | None, UtcDatetime | None]` as input and then parse results to a string?

Reply:
The `field_validator` defaults to `mode="after"`. By the time this method runs Pydantic has already ensured it is a string. Setting it up to accept other possibilities would require changing the field definition on the model. Which in turn would change what the API accepts, and the output spec. While I do think it would be nice to allow those other options, they don't really comply with the spec. And not sure there is any way to handle anything else cleanly.